src/frontend/screens/Settings/Config.tsx

```
@@ -58,11 +69,25 @@ export const Config: NativeNavigationComponent<'Config'> = ({navigation}) => {
     return () => {
       unsubscribe();
     };
-  }, [navigation, selectAndImportConfigMutation.isPending]);
+  }, [navigation, configImportIsPending]);
 
-  async function importConfigFile() {
+  async function selectAndImportConfigFile() {
     if (!isCoordinator) return;
-    selectAndImportConfigMutation.mutate();
+
+    selectFileMutation.mutate(
+      {
+        copyToCacheDirectory: true,
+        extensionFilters: ['comapeocat', 'zip'],
```

Is there any room here to check the contents of the zip files (in file-system.ts)? to validate that once they are unzipped they are the correct file type? Otherwise it just fails silently, right?

---

unfortunately not really. when using the config import mutation that actually calls the backend, the backend will check the contents and throw an error if invalid, which is propagated to the error callback of the mutation and can be handled there.